lib/OpenQA/Task/Needle/Save.pm

```diff
@@ -101,10 +102,13 @@ sub _save_needle {
 
     # ensure needle dir is up-to-date
     my $git = OpenQA::Git->new({app => $app, dir => $needledir, user => $user});
+    # avoid retrying if cleanup via Git is not configured (then an admin might need to
+    # manually cleanup the repo first)
+    $signal_guard->abort(1) if !$git->enabled || $git->config->{do_cleanup} eq 'no';
```

Comment:
The only caveat with this is that the job will just be marked as normally finished in case of a signal. A failing job might be better to get an admins attention. On the other hand, on a setup without cleanup enabled the admin is probably also just the user clicking the save button and working otherwise closely with the repo (in a manual way). So I guess this is fine.